cli/lib/kontena/cli/nodes/ssh_command.rb

```diff
@@ -4,28 +4,47 @@ class SshCommand < Kontena::Command
     include Kontena::Cli::GridOptions
 
     parameter "NODE_ID", "Node id"
+    parameter "[COMMANDS] ...", "Run command on host"
+
     option ["-i", "--identity-file"], "IDENTITY_FILE", "Path to ssh private key"
     option ["-u", "--user"], "USER", "Login as a user", default: "core"
     option "--private-ip", :flag, "Connect to node's private IP address"
     option "--internal-ip", :flag, "Connect to node's internal IP address (requires VPN connection)"
 
+    requires_current_master
+    requires_current_grid
+
     def execute
-      require_api_url
-      require_current_grid
-      token = require_token
+      node = client.get("nodes/#{current_grid}/#{node_id}")
+
+      provider = node["labels"].find{ |l| l.start_with?('provider=')}.to_s.split('=').last
```

Comment:
if find returns nil, `nil.to_s` results in `""`, `"".split('=')` results in `[]`, ´[].last` results in nil.

a bit magical yes.